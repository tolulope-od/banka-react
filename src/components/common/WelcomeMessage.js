import React from 'react';
import PropTypes from 'prop-types';

const WelcomeMessage = ({ messageHead, messageBody, optionalLink }) => {
  return (
    <div className="col s5">
      <h1 className="tag-heading">{messageHead}</h1>
      <p className="bodyTxt intro">{messageBody}</p>
      {optionalLink}
    </div>
  );
};

WelcomeMessage.propTypes = {
  messageHead: PropTypes.string.isRequired,
  messageBody: PropTypes.string.isRequired,
  optionalLink: PropTypes.object
};

export default WelcomeMessage;
