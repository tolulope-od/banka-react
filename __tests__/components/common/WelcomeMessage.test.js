import React from 'react';
import { Link } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WelcomeMessage from '../../../src/components/common/WelcomeMessage';

describe('<WelcomeMessage />', () => {
  it('Renders the WelcomeMessage component', () => {
    const wrapper = shallow(
      <WelcomeMessage
        messageHead="Banking, done right"
        messageBody="Banka is the future of bank accounts management. We employ cutting edge technology
to improve all facets of your financial life"
        optionalLink={
          <Link to="/signup" className="auth-btn-side">
            Get Started
          </Link>
        }
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();

    expect(wrapper.find('h1.tag-heading').text()).toBe('Banking, done right');
  });
});
