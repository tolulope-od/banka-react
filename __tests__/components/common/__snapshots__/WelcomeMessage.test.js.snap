// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeMessage /> Renders the WelcomeMessage component 1`] = `
<div
  className="col s5"
>
  <h1
    className="tag-heading"
  >
    Banking, done right
  </h1>
  <p
    className="bodyTxt intro"
  >
    Banka is the future of bank accounts management. We employ cutting edge technology
to improve all facets of your financial life
  </p>
  <Link
    className="auth-btn-side"
    to="/signup"
  >
    Get Started
  </Link>
</div>
`;
